pkg/ssh/ssh.go

```diff
@@ -50,22 +51,30 @@ func SetAuthorizedKeys(cfg *config.CloudConfig, withNet bool) error {
 }
 
 func getKey(key string, withNet bool) (string, error) {
-	if !strings.HasPrefix(key, "github:") {
+	providers := map[string]string{
+		"github": "https://github.com/%s.keys",
+		"gitlab": "https://gitlab.com/%s.keys",
+	}
+
+	url, err := url.Parse(key)
+	if err != nil {
+		return "", err
+	} else if url.Scheme == "" {
 		return key, nil
 	}
 
 	if !withNet {
 		return "", nil
+
+	if providerURL, ok := providers[url.Scheme]; ok {
+		key = fmt.Sprintf(providerURL, url.Opaque)
 	}
 
-	var (
-		resp *http.Response
-		err  error
-	)
+	var resp *http.Response
 	for i := 0; i < 10; time.Sleep(time.Second) {
 		// network interface(s) can be up before DNS is ready, so let's try up to 10 times
-		resp, err = http.Get(fmt.Sprintf("https://github.com/%s.keys", strings.TrimPrefix(key, "github:")))
-		if err == nil {
+		resp, err = http.Get(key)
+		if err == nil || strings.Contains(err.Error(), "unsupported protocol scheme") {
```

Comment:
With my suggestions above, this content check on the error is not necessary.

Reply:
See my previous comment about this.. but also If we were to remove this then if someone provided an ftp url then we would retry 10 times before we error out, which is why I added this check to error out early if the url is unsupported.